tests/console/login.go

```diff
@@ -116,11 +126,9 @@ func LoginToFedora(vmi *v1.VirtualMachineInstance) error {
 	// Do not login, if we already logged in
 	b := append([]expect.Batcher{
 		&expect.BSnd{S: "\n"},
-		&expect.BExp{R: PromptExpression},
-		&expect.BSnd{S: "echo $?\n"},
-		&expect.BExp{R: RetValue("0")},
+		&expect.BExp{R: fmt.Sprintf(`(\[fedora@(localhost|%s) ~\]\$ |\[root@(localhost|%s) fedora\]\# )`, vmi.Name, vmi.Name)},
```

Comment:
I guess cirros needs a similar fix. Unless it blocks this PR I believe the login part deserves a separate PR.

Reply:
https://github.com/kubevirt/kubevirt/pull/6181

Reply:
On Cirros there is no issue, because the prompt of both sudo and non sudo are just `$` and this is what it waits

if `$` was printed in the middle of the boot it was an issue, but then we would saw it before already